tests/tests_pytorch/callbacks/test_device_stats_monitor.py

```diff
@@ -166,3 +168,45 @@ def test_device_stats_monitor_warning_when_psutil_not_available(monkeypatch, tmp
     # TODO: raise an exception from v1.9
     with pytest.warns(UserWarning, match="psutil` is not installed"):
         monitor.setup(trainer, Mock(), "fit")
+
+
+def test_device_stats_monitor_logs_for_different_stages(tmpdir):
+    """Test that metrics are logged for all stages that is training, testing and validation."""
+
+    model = BoringModel()
+    device_stats = DeviceStatsMonitor()
+
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        max_epochs=2,
+        limit_train_batches=7,
+        log_every_n_steps=1,
+        accelerator="cpu",
+        devices=1,
+        callbacks=[device_stats],
+        logger=CSVLogger(tmpdir),
+        enable_checkpointing=False,
+        enable_progress_bar=False,
+    )
+
+    trainer.fit(model)
+    trainer.test(model)
+
+    with open(f"{tmpdir}/lightning_logs/version_0/metrics.csv") as csvfile:
+
+        content = csv.reader(csvfile, delimiter=",")
+        it = iter(content).__next__()
+
+        # searching for training stage logs
+        train_stage_results = [re.match(r".+on_train_batch", i) for i in it]
+        train = any(train_stage_results)
```

Comment:
Can you run the assertions after each stage instead of all together at the end? It's easier to know what failed if it does then

```suggestion
        assert any(train_stage_results)
```